python/lsst/ts/wep/GenerateDonutCatalogOnlineTask.py

```
@@ -0,0 +1,169 @@
+
+
+class GenerateDonutCatalogOnlineTask(pipeBase.PipelineTask):
+    """
+    Create a WCS from boresight info and then use this
+    with a reference catalog to select sources on the detectors for AOS.
+    """
+
+    ConfigClass = GenerateDonutCatalogOnlineTaskConfig
+    _DefaultName = "GenerateDonutCatalogOnlineTask"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.filterName = self.config.filterName
+        self.boresightRa = self.config.boresightRa
+        self.boresightDec = self.config.boresightDec
+        self.boresightRotAng = self.config.boresightRotAng
+        self.cameraName = self.config.cameraName
+
+        # TODO: Temporary until DM-24162 is closed at which point we
+        # can remove this
+        os.environ["NUMEXPR_MAX_THREADS"] = "1"
+
+    def filterResults(self, resultsDataFrame):
+        """
+        Here is where we will set up specifications for the sources
+        we want to use (i.e., filter on magnitude, blended, etc.).
+        For now it just returns all sources.
+        """
+
+        return resultsDataFrame
+
+    def run(
+        self,
+        refCatalogs: typing.List[afwTable.SimpleCatalog],
+    ) -> pipeBase.Struct:
+
+        refObjLoader = ReferenceObjectLoader(
+            dataIds=[ref.dataId for ref in refCatalogs],
+            refCats=refCatalogs,
+            config=LoadReferenceObjectsConfig(),
+        )
+        # This removes the padding around the border of detector BBox when
+        # matching to reference catalog.
+        # We remove this since we only want sources within detector.
+        refObjLoader.config.pixelMargin = 0
+
+        # Set up pandas dataframe
+        fieldObjects = pd.DataFrame([])
+        ra = []
+        dec = []
+        centroidX = []
+        centroidY = []
+        det_names = []
+
+        # Get camera. Only 'lsstCam' for now.
+        if self.cameraName == "lsstCam":
```

I think it is better to use the `Utility.CamType` and the user could put this in the runtime or read from the configuration file (`camMapper` in `policy/default.yaml`). You could raise the error if that camera is not supported yet. It is better to not hard-coded the camera here.

---

I would rather have it come from the `GenerateDonutCatalogOnlineTaskConfig` class so we can keep everything for this Task in one place. I think this will be especially useful when running as a pipeline task since the configurations can all be specified in the pipeline `yaml` file used to specify the tasks and configurations in a pipeline.

---

Yes. I think you are right. This comes from `self.cameraName = self.config.cameraName`. Thanks!